src/transaction/transaction.rs

```diff
@@ -670,11 +700,62 @@ impl Transaction {
     fn is_pessimistic(&self) -> bool {
         matches!(self.options.kind, TransactionKind::Pessimistic(_))
     }
+
+    async fn spawn_heartbeat(&mut self) {
+        let c_status = self.status.clone();
+        let primary_key = match self.buffer.get_primary_key().await {
+            Some(k) => k,
+            None => panic!("should have primary key"),
+        };
+        let start_ts = self.timestamp.clone();
+        let resolve_lock = self.options.retry_options.lock_backoff.clone();
+        let retry_region = self.options.retry_options.region_backoff.clone();
+        let rpc = self.rpc.clone();
+
+        let task = async move {
+            loop {
+                tokio::time::sleep(DEFAULT_HEARTBEAT_INTERVAL).await;
+                {
+                    let status = c_status.read().await;
+                    if matches!(
+                        *status,
+                        TransactionStatus::Rolledback | TransactionStatus::Committed
```

Comment:
When a txn gets dropped, it will never become `Rolledback` or `Committed`. We should stop heartbeat when it happens.